misc/experimental/rust/celix/src/log_helper.rs

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+use std::sync::{Arc, Mutex, RwLock};
+
+use super::BundleContext;
+use super::LogLevel;
+
+use celix_bindings::celix_log_service_t;
+use crate::ServiceTracker;
+
+pub struct LogHelper {
+    name: String,
+    tracker: Mutex<Option<ServiceTracker<celix_log_service_t>>>,
```

Comment:
With `LogHelper` as the sole owner, the use of `Mutex` is weird, whose purpose is to suppress compiler's complain of `helper.tracker.lock().unwrap().replace(tracker)`?